test/integration/protections/MentionFloodTest.ts

```diff
@@ -0,0 +1,187 @@
+            });
+        } finally {
+            await moderator.stop();
+        }
+
+    });
+
+    // Tests that regular spam isnt triggering this
+    it('Regular flooding should not be detected by MentionFlood Protection', async function () {
+        // Create users
+        const mjolnir = config.RUNTIME.client!;
+        const mjolnirUserId = await mjolnir.getUserId();
+        const moderator = await newTestUser({ name: { contains: "moderator" } });
+        const userA = await newTestUser({ name: { contains: "a" } });
+
+        this.moderator = moderator;
+        this.userA = userA;
+
+        // Join managementroom and setup target room
+        await moderator.joinRoom(config.managementRoom);
+        LogService.debug("MentionFloodTest", `Joining managementRoom: ${config.managementRoom}`);
+        let targetRoom = await moderator.createRoom({ invite: [mjolnirUserId] });
+        LogService.debug("MentionFloodTest", `moderator creating targetRoom: ${targetRoom}; and inviting ${mjolnirUserId}`);
+        await moderator.sendMessage(this.mjolnir.managementRoomId, { msgtype: 'm.text', body: `!mjolnir rooms add ${targetRoom}` });
+        LogService.debug("MentionFloodTest", `Adding targetRoom: ${targetRoom}`);
+        try {
+            await moderator.start();
+            await userA.start();
+            await moderator.sendMessage(this.mjolnir.managementRoomId, { msgtype: 'm.text', body: `!mjolnir enable MentionFlood` });
+            LogService.debug("MentionFloodTest", `Enabling MentionFlood protection`);
+
+            // Idea here is that we send spam to the room and then get the mjolnir status as a test case.
+            // If everything goes well we should only see the status in the management room and no spam warnings.
+            const statusResp = await getFirstMessage(this.mjolnir, this.mjolnir.managementRoom, async () => {
+                await produceRandomSpam(userA, targetRoom);
+                return await moderator.sendMessage(this.mjolnir.managementRoom, { msgtype: 'm.text', body: `!mjolnir` });
+            });
+            assert.equal(statusResp.content.body.includes('Protected rooms: 1'), true, 'unexpected message from mjolnir while fake spamming');
+        } finally {
+            await moderator.stop();
+            await userA.stop();
+        }
+    });
+
+    // Tests that normal mentions do not trigger the protection
+    it('Regular mentions should not be detected by MentionFlood Protection', async function () {
+        // Create users
+        const mjolnir = config.RUNTIME.client!;
+        const mjolnirUserId = await mjolnir.getUserId();
+        const moderator = await newTestUser({ name: { contains: "moderator" } });
+        const userA = await newTestUser({ name: { contains: "a" } });
+
+        this.moderator = moderator;
+        this.userA = userA;
```

Comment:
nitpick but `userA` doesn't tell us what this use is doing. they could unrelated to moderation of the room, they could just be a participant, they could even be a user who is going to be spamming, who is `userA`?